Integracion de ListadoAlumnosPorMateria

gregorionunez8 opened the pull request:
- Modifique InscripcionData para que me envie todas las materias sin importar el estado(ni de materia,ni de alumno)
- Cree un metodo en MateriaData para enviarle un String con el nombre de la materia y me devuelva el idMateria
- En Menu agregue un MenuItem en consulta para abrir el JInternalFrame de ListadoDeAlumnosPorMateria
- Integre ListadoDeAlumnosPorMateria con el BE
ListadoDeAlumnosPorMateria:
Busca alumnos por materia (Testeado)
RadioButton para incluir materias inactivas (Testeado)
RadioButton para incluir alumnos inactivos (Testeado)
Button para salir (Testeado)

gregorionunez8 committed:
Integracion de ListadoAlumnosPorMateria

- Modifique InscripcionData para que me envie todas las materias sin importar el estado(ni de materia,ni de alumno)
- Cree un metodo en MateriaData para enviarle un String con el nombre de la materia y me devuelva el idMateria
- En Menu agregue un MenuItem en consulta para abrir el JInternalFrame de ListadoDeAlumnosPorMateria
- Integre ListadoDeAlumnosPorMateria con el BE
ListadoDeAlumnosPorMateria:
Busca alumnos por materia (Testeado)
RadioButton para incluir materias inactivas (Testeado)
RadioButton para incluir alumnos inactivos (Testeado)
Button para salir (Testeado)

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/InscripcionData.java:
```diff
@@ -214,11 +214,11 @@ public void actualizarNota(int idAlumno, int idMateria, double nota){
                       "INNER JOIN inscripcion ON materia.idMateria = inscripcion.idMateria " +
                       "INNER JOIN alumno ON alumno.idAlumno = inscripcion.idAlumno " +
-                      "WHERE materia.estado=1 AND alumno.estado=1 AND materia.idMateria=?";
+                      "WHERE materia.idMateria=?";
```
Por que querrias listar alumnos inactivos ?
Por que querrias saber los alumnos que estan en una materia inactiva?

gregorionunez8 replied:
Para de alguna forma poder acceder a toda la informacion de la base de datos, sino seria necesario entrar a la base de datos mediante phpadmin. Si no lo ven necesario, lo borro.

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -131,12 +131,11 @@ public List listarMaterias() {
             while (rs.next()) { // Bucle para agregar elementos a la lista
                 lista.add(new Materia(rs.getInt("idMateria"), rs.getString("nombre"), rs.getInt("anio"), rs.getBoolean("estado")));
             }
-            JOptionPane.showMessageDialog(null, "Listado de materias completo"); // Mensaje
```
Este cambio es correcto?

gregorionunez8 replied:
Si, porque sino cada vez que entras al JInternalFrame de ListadosDeAlumnosPorMateria saldria el mensaje

MaribelAngulo commented on UniversidadGrupo67/src/AccesoADatos/MateriaData.java:
```diff
@@ -145,4 +144,25 @@ public List listarMaterias() {
+//Metodo para devolver IdMateria mediante el nombre
+    public int materiaIdPorNombre(String nombre) {
+        // Consulta sql para insertar materia
+        String sql = "SELECT idMateria FROM Materia WHERE nombre=?";
```
Que pasaria si tengo mas de una materia con el mismo nombre y distintos ids?

gregorionunez8 replied:
Te devolvería la primer coincidencia. Tengo que rellenar el ComboBox con objetos de tipo Materia y solo muestre el nombre de las materias.

MaribelAngulo commented on UniversidadGrupo67/src/newpackage/vistas/ListadoDeAlumnosPorMateria.java:
```diff
@@ -5,17 +5,47 @@
-    /**
-     * Creates new form ListadoDeAlumnosPorMateria
-     */
+    MateriaData matdata = new MateriaData();
```
renombrar matData. 
Es buena practica el uso de camelCase. Aplicar para todas las variables

gregorionunez8 replied:
Listo!

MaribelAngulo commented on UniversidadGrupo67/src/newpackage/vistas/ListadoDeAlumnosPorMateria.java:
```diff
@@ -61,89 +89,140 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
+        getContentPane().add(jScrollPane1, new org.netbeans.lib.awtextra.AbsoluteConstraints(80, 200, 590, 290));
+
+        jRBAlumnos.setFont(new java.awt.Font("Tahoma", 2, 14)); // NOI18N
+        jRBAlumnos.setText("Incluir inactivos");
```
Agregaste una nueva funcionalidad ?

gregorionunez8 replied:
Si, un RadioButton para incluir los alumnos inactivos de esa materia.

MaribelAngulo commented on UniversidadGrupo67/src/newpackage/vistas/ListadoDeAlumnosPorMateria.java:
```diff
@@ -61,89 +89,140 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
+    //Metodo para rellenar la tabla
+    public void rellenarTabla() {
+        String nombreMateria = (String) jCBMateria.getSelectedItem(); // Extraigo el String del ComboBox
+        int idMateria = matdata.materiaIdPorNombre(nombreMateria); //Extraigo el idMateria usando el metodo de MateriaData enviandole el nombre
```
El id de la materia lo puedes sacar del combobox, no es necesario crear un metodo.
Coloca esto en el action performed del combobox:
`Materia materiaSeleccionada = (Materia) jCBMateria.getSelectedItem();`
Luego tu idMateria la obtienes asi:
`materiaSeleccionada.getIdMateria()`

gregorionunez8 replied:
No, porque lo que devuelve jCBMateria.getSelectedItem() es un objeto con solo el nombre de la materia

MaribelAngulo replied:
por que no muestras en el combobox el nombre de la materia y el codigo? 
Si solo muestras nombres de materias pueden repetirse los nombres, los codigos no
Si tienes la materia y el codigo en el combobox podrias aplicar las lineas de codigo antes mencionadas

gregorionunez8 replied:
Listo!

MaribelAngulo commented on UniversidadGrupo67/src/newpackage/vistas/ListadoDeAlumnosPorMateria.java:
```diff
@@ -61,89 +89,140 @@ public void actionPerformed(java.awt.event.ActionEvent evt) {
+    public void rellenarTabla() {
+        String nombreMateria = (String) jCBMateria.getSelectedItem(); // Extraigo el String del ComboBox
+        int idMateria = matdata.materiaIdPorNombre(nombreMateria); //Extraigo el idMateria usando el metodo de MateriaData enviandole el nombre
+        ArrayList<Alumno> alumno = new ArrayList<Alumno>(insdata.obtenerAlumnosXMateria(idMateria)); //Hago un ArrayList con los alumnos
```
renombrar alumnos
Cuando las variables tienen varios valores como las listas es buena practica darles nombres en plural
Aplicar para todos los casos

gregorionunez8 replied:
Listo!

gregorionunez8 committed:
Correccion de ListadoAlumnosPorMateria

- Elimine metodo para buscar idMateria por nombre
- Modifique el toString de la clase Materia para que en el combo box me muestre solo el nombre y año
- Corregi nombre de variables

gregorionunez8 committed:
Agregue ID
